syntaxes/haskell.YAML-tmLanguage

```diff
@@ -647,9 +677,19 @@ repository:
       - include: '#comments'
     end: |
       (?x)
-        (?=               # we look ahead, but we do not want to consume
-          (<\-)           # we are the left side of a `x :: Type <- expr` bind statement
-          | }             # A block closed? Maybe this should also include `;`, because non-indentation based `do`
+        (?= 
+          # we look ahead, but we do not want to consume
+
+            # are we to the left side of a symbolic keyword such as = or <- ?
+              # non-symbolic character
+              (?<![\p{S}\p{P}&&[^(),;\[\]`{}_"']])
+              # symbolic keyword except (->)
+              ((<-)|(=)|(-<)|(-<<))
+              # non-symbolic character
+              ([(),;\[\]`{}_"']|[^\p{S}\p{P}])
+
+            # A block closed? Maybe this should also include `;`, because non-indentation based `do`
+            |}
```

Comment:
This adds a check that the symbolic operator is not surrounded by other symbols (to avoid mistakenly catching `<--` as a reserved do arrow for instance), and adds a few other reserved operators other than `(<-)`.

Reply:
Oh my, yeah that's a pretty complex regex. Makes sense though and yes, I agree, `;` probably should be in there.

Reply:
Haha you were the one to write that comment.

I admit I have not paid much attention to how things work with `{;}` layout, it seems you would need `begin`..`end` to handle those situations. Unless you have a specific suggestion, I think for the time being I won't address this; we can think about issues with `{;}` layout later.